Improve SQL file connector documentation (#2794)
hazelcast/hazelcast-jet: site/docs/sql/file-connector.md

`/path/to/directory/`.

`glob` is a pattern to filter the files in the specified directory.
The default value is `*`, matching all files.

### Ignore File Not Found

> When you create a mapping without a schema, the location specified by the

Reviewer: The term "schema" is quite confusing in sql. It's an actual object in some DBs, in calcite it denotes a hierarchical collection of objects etc. I'd rather avoid it and use this:
```suggestion
When you create a mapping without a column list, the location specified by the
```

`path` option is expected to contain at least one file matching the
`glob`, otherwise an exception is thrown. This is to avoid hard to catch

> mistakes, such as typos. If you simply want to return zero results set

Reviewer: ```suggestion
mistakes, such as typos. If you simply want to return zero results, set
```

the `ignoreFileNotFound` option to `true`. Note that in this case you

> must specify the schema.

Reviewer: ```suggestion
must specify the column list.
```

This option is not valid for [file table functions](#file-table-functions),

> because they always need at least one record to derive the schema from.

Reviewer: ```suggestion
because they always need at least one record to derive the column list from.
```

The default value is `false`.

> ### Shared File System option

Reviewer: ```suggestion
### Shared File System Option
```
